src/main/java/com/cleanroommc/groovyscript/compat/mods/tinkersconstruct/EntityMelting.java

```diff
@@ -0,0 +1,164 @@
+        restoreFromBackup().forEach(recipe -> TinkerRegistryAccessor.getEntityMeltingRegistry().put(recipe.name, recipe.result));
+    }
+
+    @MethodDescription(type = MethodDescription.Type.QUERY)
+    protected List<EntityMeltingRecipe> getAllRecipes() {
+        return TinkerRegistryAccessor.getEntityMeltingRegistry().entrySet().stream().map(EntityMeltingRecipe::fromMapEntry).collect(Collectors.toList());
+    }
+
+    public EntityMeltingRecipe add(EntityEntry entity, FluidStack output) {
+        EntityMeltingRecipe recipe = new EntityMeltingRecipe(entity, output);
+        add(recipe);
+        return recipe;
+    }
+
+    public void add(EntityMeltingRecipe recipe) {
+        if (recipe == null || recipe.name == null) return;
+        addScripted(recipe);
+        TinkerRegistryAccessor.getEntityMeltingRegistry().put(recipe.name, recipe.result);
+    }
+
+    public boolean remove(EntityMeltingRecipe recipe) {
+        if (recipe == null || recipe.name == null) return false;
+        addBackup(recipe);
+        TinkerRegistryAccessor.getEntityMeltingRegistry().remove(recipe.name, recipe.result);
+        return true;
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByInput")
+    public boolean removeByInput(EntityEntry entity) {
+        ResourceLocation name = entity.getRegistryName();
+        if (TinkerRegistryAccessor.getEntityMeltingRegistry().entrySet().removeIf(entry -> {
+            boolean found = entry.getKey().equals(name);
+            if (found) addBackup(new EntityMeltingRecipe(entry.getKey(), entry.getValue()));
+            return found;
+        })) return true;
+
+        GroovyLog.msg("Error removing Tinkers Construct Entity Melting recipe")
+                .add("could not find recipe with input {}", name)
+                .error()
+                .post();
+        return false;
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByOutput")
+    public boolean removeByOutput(FluidStack output) {
+        if (TinkerRegistryAccessor.getEntityMeltingRegistry().entrySet().removeIf(entry -> {
+            boolean found = entry.getValue().isFluidEqual(output);
+            if (found) addBackup(new EntityMeltingRecipe(entry.getKey(), entry.getValue()));
+            return found;
+        })) return true;
+
+        GroovyLog.msg("Error removing Tinkers Construct Entity Melting recipe")
+                .add("could not find recipe with output {}", output)
+                .error()
+                .post();
+        return false;
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeAll", priority = 2000, example = @Example(commented = true))
+    public void removeAll() {
+        TinkerRegistryAccessor.getEntityMeltingRegistry().forEach((name, result) -> addBackup(new EntityMeltingRecipe(name, result)));
+        TinkerRegistryAccessor.getEntityMeltingRegistry().forEach(TinkerRegistryAccessor.getEntityMeltingRegistry()::remove);

```

Comment:
why does `clear` not work? currently, running any `removeAll` from tinkers generates a CME.
```suggestion
        TinkerRegistryAccessor.getEntityMeltingRegistry().clear();
```

Reply:
this change should also be applied to all classes